crates/sui-rpc-benchmark/src/bin/pull_grafana_logs.rs

```diff
@@ -38,14 +38,52 @@ struct GrafanaLog {
     message: String,
 }
 
-fn extract_body_from_message(message: &str) -> Option<String> {
+#[derive(Debug)]
+struct LogEntry {
+    timestamp: String,
+    host: String,
+    method: String,
+    body: String,
+}
+
+/// One example message is:
+/// 2025-02-11T23:15:17.944697206Z stderr F 2025-02-11T23:15:17.944501Z  INFO sui_edge_proxy::handlers: Sampled read request headers={"host": "wallet-rpc.mainnet.sui.io", "client-sdk-type": "typescript", "client-sdk-version": "1.17.0", "client-target-api-version": "1.40.0", "client-request-method": "suix_getBalance", "content-type": "application/json", "content-length": "152", "accept-encoding": "gzip", "user-agent": "okhttp/4.9.2", "x-cloud-trace-context": "31caa7db658044d850a002ccf4ff15b1/8018737809747708392", "cookie": "_cfuvid=h0GD1bYot45Ln6kVCdL4qsFCCyw3h2cLw3caDNmhWNw-1739262948231-0.0.1.1-604800000", "via": "1.1 google", "x-forwarded-for": "171.236.184.3, 34.8.28.138", "x-forwarded-proto": "https", "connection": "Keep-Alive"} body=b"{\"jsonrpc\":\"2.0\",\"id\":189393,\"method\":\"suix_getBalance\",\"params\":[\"0x23cad599a375b9c2cedd62fa20112526c90a71764230425cb7f557c0c0b3b150\",\"0x2::sui::SUI\"]}" peer_type=Read
```

Comment:
This is very helpful for understanding what's going on in this function, thank you -- I'm slightly horrified now that I understand the format being parsed, but we can leave that as a problem for another day.